docs/development.rst

```diff
@@ -17,268 +34,289 @@ reflected without any action on your part. Useful vagrant sub-commands::
+    docker-compose up -d             # Start the containers
+    docker-compose rm                # Destroy the containers
+
+There are ``make`` shortcuts on the host for frequent commands, such as::
+
+    make up         # docker-compose up -d
+    make bash       # docker exec -it kuma_web_1 bash
+    make shell_plus # docker exec -it kuma_web_1 ./manage.py shell_plus
+
+Run all commands in this doc in the ``kuma_web_1`` container after ``make bash``
 
 Running Kuma
 ============
-
-You can start all Kuma servers and services on the vm with::
+The Vagrant environment runs everything in a single VM. It runs MySQL,
+ElasticSearch, Apache, and other "backend" services whenever the VM is running.
+There are additional Kuma-specific services that are configured in
+``Procfile``, and are run with::
 
     foreman start
 
+The development instance is then available at https://developer-local.allizom.org.
+
+When the Docker container environment is started (``make up`` or similar), all
+of the services are also started. The development instance is available at
+http://localhost:8000.
+
 Running the Tests
 =================
-
 A great way to check that everything really is working is to run the test
 suite.
 
 Django tests
 ------------
-
-Running the kuma Django test suite is easy::
+Run the Django test suite::
 
     make test
 
 For more information, see the :doc:`test documentation <tests>`.
 
 Front-end tests
 ---------------
-
 To run the front-end (selenium) tests, see :doc:`Client-side Testing with
 Intern <tests-ui>`.
 
 Kumascript tests
 ----------------
-
 If you're changing Kumascript, be sure to run its tests too.
 See https://github.com/mozilla/kumascript
 
 Compiling Stylus Files
 ======================
+Stylus files need to be compiled for changes to take effect.
 
-Stylus files need to be compiled for changes to take effect. Vagrant will
-automatically compile Stylus files when they change, but compilation can also be
-run manually::
+In the Vagrant environment, the ``foreman`` task ``stylus`` will automatically
+compile Stylus files when they change, placing the generated CSS files at
+``build/assets/css``.
 
-    compile-stylesheets
+In either environment, compilation can be run manually::
 
-The relevant CSS files will be generated and placed within the
-`build/assets/css` directory. You can add a ``-w`` flag to that call to compile
-stylesheets upon save.
+    scripts/compile-stylesheets
 
-Database Migrations
-===================
+To watch for changes to the files and recompile::
+
+    scripts/compile-stylesheets -w
 
-Basically all apps are migrated using Django's migration system.
+Watching for file changes performs well in the Vagrant environment, but can be
+slow with the host-mounted files in the Docker container.
 
-See the Django documentation for the
-`migration workflow <https://docs.djangoproject.com/en/1.8/topics/migrations/#workflow>`_.
+.. _fswatch: https://github.com/emcrisostomo/fswatch#getting-fswatch
 
```

Comment:
This link isn't actually used, did you intend to add some docs about fswatch in this PR or did you decide to delay that for a future PR and just miss this link?